### PR TITLE
Fix error page rendering with unclosed Output buffers

### PR DESCRIPTION
**Pull request by Fedik**
### Summary of Changes

In some cases the error page is rendered incorrectly, when exception happen while rendering.

### Testing Instructions
Applly patch.
Add an error `throw new \Exception('That an error');` in to `layouts/joomla/content/info_block/hits.php`
On the site open the article detail view.

Also can try to add an error in to the text field layout `layouts/joomla/form/field/text.php`
And try to edit the Article.

### Actual result BEFORE applying this Pull Request
You get broken page:
Half article are rendered,
And then an error is rendered

### Expected result AFTER applying this Pull Request
Only error is rendered

### Link to documentations
Please select:
- [x] No documentation changes for docs.joomla.org needed
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: libraries/src/Exception/ExceptionHandler.php [hunk 1]
```diff
@@ -94,6 +94,11 @@ public static function render(\Throwable $error)
                 $app->redirect('index.php');
             }
 
+            // Clear all opened Output buffers to prevent misrendering
+            for ($i = 0, $l = ob_get_level(); $i < $l; $i++) {
```
Comment (Denitz) [inline libraries/src/Exception/ExceptionHandler.php:98]:
```suggestion
            while (ob_get_level()) {
```

**Comment by wilsonge** [comment]
I think it would be good in core for us to close the buffers using the finally keyword. So run a try/finally (you don't actually need to have a catch) and putting this in the Layout, HtmlView classes (does Document use it too when we include the template?) would also be a smart idea I guess.

I mean I'm not against this but this should be a method of last resort rather than the go to thing.

File: libraries/src/Exception/ExceptionHandler.php (hunk 1 above)
Comment (Fedik) [reply libraries/src/Exception/ExceptionHandler.php:98]: I would prefer my version, because it reads ob_get_level() once, instead of in every loop.

**Comment by Fedik** [comment]
@wilsonge I not sure that it is possible, we do not have try blocks everywhere,
it can require more work, to review every possible place.

**Comment by Fedik** [comment]
Okay, I will keep it as it is. There 100+ places where we use output buffering, some are in the layouts file.
It is more practical to close all at once before error rendering.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 52d1b9463f3fbf0d14d1684a355c68ec503cc17e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41188">issues.joomla.org/tracker/joomla-cms/41188</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 52d1b9463f3fbf0d14d1684a355c68ec503cc17e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41188">issues.joomla.org/tracker/joomla-cms/41188</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41188">issues.joomla.org/tracker/joomla-cms/41188</a>.</sub>

**Comment by obuisard** [comment]
Thank you Fefir @Fedik for the PR

**Pull request merged**